adapters/handlers/rest/handlers_snapshot.go

```diff
@@ -0,0 +1,167 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package rest
+
+import (
+	middleware "github.com/go-openapi/runtime/middleware"
+	"github.com/semi-technologies/weaviate/adapters/handlers/rest/operations"
+	"github.com/semi-technologies/weaviate/adapters/handlers/rest/operations/schema"
+	"github.com/semi-technologies/weaviate/adapters/handlers/rest/state"
+	"github.com/semi-technologies/weaviate/adapters/repos/db"
+	"github.com/semi-technologies/weaviate/entities/models"
+	entitySchema "github.com/semi-technologies/weaviate/entities/schema"
+	"github.com/semi-technologies/weaviate/entities/snapshots"
+	"github.com/semi-technologies/weaviate/usecases/auth/authorization/errors"
+	"github.com/semi-technologies/weaviate/usecases/backup"
+	schemaUC "github.com/semi-technologies/weaviate/usecases/schema"
+	"github.com/semi-technologies/weaviate/usecases/sharding"
+)
+
+func newSnapshotterProvider(db *db.DB) backup.SnapshotterProvider {
+	return &snapshotterProvider{db}
+}
+
+type snapshotterProvider struct {
+	db *db.DB
+}
+
+func (sp *snapshotterProvider) Snapshotter(className string) backup.Snapshotter {
+	if idx := sp.db.GetIndex(entitySchema.ClassName(className)); idx != nil {
+		return idx
+	}
+	return nil
+}
+
+type backupHandlers struct {
+	manager *backup.Manager
+}
+
+func (s *backupHandlers) createSnapshot(params schema.SchemaObjectsSnapshotsCreateParams,
```

Comment:
should this be named `createBackup` at this level, since it is request handler logic, or are we considering everything beneath the initial request for backup as "snapshot"?

I think we briefly discussed that user-facing code should reference "backup" and internal logic should be "snapshot", but I'm not clear on where this line is drawn.

Reply:
I agree.
I tried to not rename it to make the review as smoother as possible.
Later on everything related to snapshot should be renamed to backup